src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -823,19 +2357,280 @@ private void DisplayNodeContextMenu(object sender, RoutedEventArgs e)
 
             // Clearing any existing items in the node's ContextMenu.
             contextMenu.Items.Clear();
-            NodeContextMenuBuilder.Build(contextMenu, viewModel, NodeViewCustomizationMenuItems);
+            nodeContextMenu.Items.Clear();
+            NodeContextMenuBuilder.Build(nodeContextMenu, viewModel, NodeViewCustomizationMenuItems);
 
-            contextMenu.DataContext = viewModel;
-            contextMenu.IsOpen = true;
+            nodeContextMenu.DataContext = viewModel;
+            nodeContextMenu.Closed += MainContextMenu_OnClosed;
+            nodeContextMenu.IsOpen = true;
 
             e.Handled = true;
         }
 
         private void MainContextMenu_OnClosed(object sender, RoutedEventArgs e)
         {
-            grid.ContextMenu.Items.Clear();
+            nodeContextMenu.Closed -= MainContextMenu_OnClosed;
+            nodeContextMenu.Items.Clear();
             e.Handled = true;
         }
 
+        private static Style GetContextMenuStyle()
```

Comment:
Should we create a separate static style repository (class) as these are shared styles?